crates/shared/src/price_estimation/oneinch.rs

```diff
@@ -1,15 +1,22 @@
 use crate::{
-    oneinch_api::{OneInchClient, ProtocolCache, RestResponse, SellOrderQuoteQuery},
+    oneinch_api::{
+        OneInchClient, ProtocolCache, RestResponse, SellOrderQuote, SellOrderQuoteQuery,
+    },
     price_estimation::{
         gas, Estimate, PriceEstimateResult, PriceEstimating, PriceEstimationError, Query,
     },
+    request_sharing::RequestSharing,
 };
-use futures::StreamExt;
+use futures::{future::BoxFuture, FutureExt, StreamExt};
 use model::order::OrderKind;
 use std::sync::Arc;
 
 pub struct OneInchPriceEstimator {
     api: Arc<dyn OneInchClient>,
+    sharing: RequestSharing<
```

Comment:
One thing that I'm not super fond of is that each estimator needs its own `sharing` instance right?

Would't you also be able to implement a `BufferedPriceEstimator` around another `SinglePriceEstimating` using the exact same helper you created? Might make composition easier (instead of relying on adding a new field on all existing estimators).

Kind of how we have a specialized `SingleOrderSolver` in the driver.